src/Common/src/Interop/Unix/System.Native/Interop.GetUnixArchitecture.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [DllImport(Libraries.SystemNative, SetLastError = true)]
+        internal static extern int GetUnixArchitecture();
+
+        internal enum ProcessorArchitecture
+        {
+            x86,
+            x64,
+            ARM
```

Comment:
Right now we are not supported Arm64, it'll be there in enum unused. When we start supporting for Arm64, we can change this since it's a OOB assembly.